lib/ansible/plugins/action/synchronize.py

```diff
@@ -33,7 +33,7 @@ class ActionModule(ActionBase):
     def _get_absolute_path(self, path):
         original_path = path
 
-        if path.startswith('rsync://'):
+        if ':' in path:
```

Comment:
I think this test is in the right direction, but it is probably a bit too broad. It should test for specific patterns at the beginning the path string to avoid false positives for strings that have a port in them or possible a colon in the path.

Reply:
In my case the path will be `user@host:artifacts/`, so the path starts with an (arbitrary) username. I was thinking of checking for the presence of `@`, but iirc I've never seen a `:` in a local file system path and I guess a `@` is more likely.
Also, `host:artifacts/` should be accept as an "absolute" path, i.e. without changing it in the `_get_absolute_path` function.

Reply:
It is possible to have a `:` in a file name on Linux filesystems (not on macOS), so assuming the presence of a colon anywhere in the path could lead to false positives.

This probably needs to be an `re.match()` test.

Reply:
Possible values are:
* `user@host:`
* `user@host:some/directory`
* `host:`
* `host:some/directory`
So a regex like `^([^@]+@)?([a-zA-Z0-9\-]+?):.*$` would match them. It's not 100% accurate though, e.g. for hostnames we could use `^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$` (source: https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address) and for usernames (dependent on the distro) and directories (dependent on the used filesystem) another long and complicated regex - but I'd suggest not going down this rabbit hole ;)

In the same file there are already a few checks similar to my suggested one:
https://github.com/ansible/ansible/blob/e9799dfd00fa87ad0d810f113435a37bfbe7533c/lib/ansible/plugins/action/synchronize.py#L76 and https://github.com/ansible/ansible/blob/e9799dfd00fa87ad0d810f113435a37bfbe7533c/lib/ansible/plugins/action/synchronize.py#L106 (i.e. without regex).

Reply:
Hmmm, that regexp does get out of hand rather quickly. Thank you for looking into that. 

I still feel testing for `:` in the path is going to result in some false positives. It's also possible to have the the port in the path: `rsync://[USER@]HOST[:PORT]/SRC`, though it looks like specifying the port only works if it starts with `rsync://`.

I'm really leery of this change but don't have a better solution.